src/algorithm/dynamics.hpp

```diff
@@ -103,10 +103,47 @@ namespace pinocchio
     a.noalias() = J.transpose() * lambda_c;
     cholesky::solve(model, data, a);
     a += data.torque_residual;
+
     
     return a;
   }
-  
+
+  ///
+  /// \brief Computes the inverse of the KKT matrix for dynamics with contact constraints, [[M JT], [J 0]].
+  /// The matrix is defined when we call forwardDynamics/impulsedynamics. This method makes use of
+  /// the matrix decompositions performed during the forwardDynamics/impulasedynamics and returns the inverse.
+  /// The jacobian should be the same that was provided to forwardDynamics/impulasedynamics.
+  /// Thus you should call forward Dynamics/impulsedynamics first.
+  /// \param[in] model The model structure of the rigid body system.
+  /// \param[in] data The data structure of the rigid body system.
+  /// \param[out] MJtJ_inv inverse of the MJtJ matrix.
+  ///
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl,
+           typename ConstraintMatrixType, typename KKTMatrixType>
+  inline void getKKTContactDynamicMatrixInverse(const ModelTpl<Scalar,Options,JointCollectionTpl> & model,
+                                                const DataTpl<Scalar,Options,JointCollectionTpl> & data,
+                                                const Eigen::MatrixBase<ConstraintMatrixType> & J,
+                                                Eigen::MatrixBase<KKTMatrixType> & MJtJ_inv)
```

Comment:
To make it work, you have to do a const_cast. See other similar functions in algo directory.

Reply:
Done.